### DISCUSSION TITLE
When does visualization decide to display things and when does it give up?

### DISCUSSION THREAD

**Discussion by hubertp** [enso-org/enso#6783] [Work In Progress - GUI]
As part of the investigation into #6766, I started to notice something weird with visualizations on startup.

Assume a simple project with a few nodes like:
![Screenshot from 2023-05-19 16-17-38](https://github.com/enso-org/enso/assets/292128/aa973c54-6ed8-48e9-80d6-6edb0ee9ed8d)
If I start the project again but with some network throttling it will show the waiting status on them for some time
![Screenshot from 2023-05-19 16-19-26](https://github.com/enso-org/enso/assets/292128/43c41948-b414-48e8-aa9f-fc085b040132)
and eventually close them 
![Screenshot from 2023-05-19 16-06-44](https://github.com/enso-org/enso/assets/292128/f5f1ddaf-1f2b-49d7-85ca-38750dbd0e8d)

The data eventually arrives but it is obviously slightly slower due to throttling.
Is this expected? If so, why? I assumed that data from LS can arrive at any time.

**Comment by farmaazon** [comment]
Visualization close themselves when some operation fails. Perhaps attaching timed out?

The console logs should show the problem.

**Reply by hubertp** [reply]
No timeouts on the LS side and data is eventually received by the browser, as indicated by inspecting binary messages received by the browser.

**Reply by farmaazon** [reply]
Given that in the ticked I just responded to (#6701) the `text/applyEdit` takes 10 seconds, I think here may be a similar problem: the ACK for attaching visualization finally arrives, but after 10 s IDE give up. Could you check if this is the case?

Also, the web console logs (after `showLogs` or with --debug option) should probably tell us why we close those visualizations.

**Reply by hubertp** [reply]
> I think here may be a similar problem: the ACK for attaching visualization finally arrives, but after 10 s IDE give up. Could you check if this is the case?
>
>Also, the web console logs (after showLogs or with --debug option) should probably tell us why we close those visualizations.

Yeap, the timeout is there, on the IDE side.

**Reply by farmaazon** [reply]
By "on the IDE side" you mean LS sends reply in time, but IDE interprets it only after 10 seconds (like in #6660)?

**Reply by hubertp** [reply]
> By "on the IDE side" you mean LS sends reply in time, but IDE interprets it only after 10 seconds (like in #6660)?

LS sends replies in time, yes. Then messages are either in transit or IDE interprets them after 10 seconds. Yes, like #6660.

**Reply by hubertp** [reply]
Alternatively, would it be wrong to just not timeout in IDE after 10 seconds on such visualizations and just continue waiting? Especially on startup?

**Reply by MichaelMauderer** [reply]
I wonder if it alternatively would be possible to revive a visualization if we receive data for it, even if we had already given up on waiting for it. 

But waiting longer would make for the better UX I guess.

**Comment by farmaazon** [comment]
Closing, as both the timeout and better reporting are already reported as issues.
